.github/workflows/ci.yml

```diff
@@ -58,6 +58,26 @@ jobs:
       - name: Checks
         run: yarn run check
 
+      - name: Set deployment url
+        run: node ./scripts/deploy/surge-domains.js url DEPLOYMENT_URL >> $GITHUB_ENV
+        env:
+          PROD: ${{ github.ref == 'refs/heads/main' }}
+
+      - name: Set deployment environment
+        run: node ./scripts/deploy/surge-domains.js env DEPLOYMENT_ENV >> $GITHUB_ENV
+        env:
+          PROD: ${{ github.ref == 'refs/heads/main' }}
+
+      - name: Create GitHub deployment
+        if: env.DEPLOYMENT_URL
+        uses: bobheadxi/deployments@v0.6.0
+        id: deployment
+        with:
+          step: start
+          token: ${{ secrets.GITHUB_TOKEN }}
+          env: ${{ env.DEPLOYMENT_ENV }}
+          ref: ${{ github.head_ref || github.ref }}
```

Comment:
`head_ref` will correctly attach a staging deployment status to a PR. Fallback to `ref` for main branch deployment statuses.